### DISCUSSION TITLE
Question about the "memory_limit" variable

### DISCUSSION THREAD

**Discussion by Amumu2020** [duckdb/duckdb#7753] [General]
My default memory_limit is around 13GB (80% of RAM).
I set it to 130MB, which is approximately 1% of the default value, and then load some tables.
Why did the "RAM-Memory Free Percent" decrease from 80% to 67%?
![1](https://github.com/duckdb/duckdb/assets/26343224/17a44775-f6f8-48fe-b70b-07cf2d219e8c)
![2](https://github.com/duckdb/duckdb/assets/26343224/d0099970-63b4-4e28-89a6-8b5eba40fb62)
My expectation is that the "RAM-Memory Free Percent" decrease from 80% to around 79%.
Is my understanding of the "memory_limit" variable incorrect?

**Comment by Tishj** [comment]
I'm not entirely sure what you're measuring, but `memory_limit` does not mean we reserve that much memory up front, it means we won't exceed that limit and will start to evict unpinned buffers once that limit is reached.

**Reply by Amumu2020** [reply]
Sorry for my poor english.
I want to find out the usage of system free memory by duckdb when loading tables.

**Test1: Loading 1G tpch**
1.  `set global memory_limit = '100MB';`
2. record the size of system free memory before loading the table: it is 12594MB
3. `call dbgen(sf = 1);`
4. record the size of system free memory after loading the table: it is 11794MB

Approximately 400MB of free memory is being use.


**Test2: Loading 5G tpch**
1.  `set global memory_limit = '100MB';`
2. record the size of system free memory before loading the table: it is 12594MB
3. `call dbgen(sf = 5);`
4. record the size of system free memory after loading the table: it is 10594MB

Approximately 2000MB of free memory is being used.

Since bufferpool consumes up to 100MB of memory, why has the system lost so much free memory?

**Reply by Tishj** [reply]
I'm not entirely sure, but it could be that the memory allocated by `dbgen` isn't managed by the buffer manager yet

**Reply by Amumu2020** [reply]
emmm... I tried `copy ... from ...`

Test3: Loading a table ( 3GB )
1. `set global memory_limit = '1GB'`;
2. record the size of system free memory before loading the table: it is 12866MB
3. `copy ods_tfc_gps from '/home/swk/gps_2000_new.csv';`
4. record the size of system free memory after loading the table: it is 4866MB

Approximately 8GB of free memory is being used.  However, bufferpool consumes up to 1GB of memory.
